crates/rpc-types/src/eth/transaction/receipt.rs

```diff
@@ -1,63 +1,65 @@
+    /// The receipt envelope, which contains the consensus receipt data..
+    #[serde(flatten)]
+    pub inner: ReceiptEnvelope,
+
     /// Transaction Hash.
     pub transaction_hash: B256,
     /// Index within the block.
-    pub transaction_index: U64,
+    #[serde(with = "alloy_serde::u64_hex_or_decimal")]
+    pub transaction_index: u64,
     /// Hash of the block this transaction was included within.
     pub block_hash: Option<B256>,
     /// Number of the block this transaction was included within.
-    pub block_number: Option<U256>,
-    /// Cumulative gas used within the block after this was executed.
-    pub cumulative_gas_used: U256,
+
+    #[serde(with = "alloy_serde::u64_hex_opt")]
+    pub block_number: Option<u64>,
     /// Gas used by this transaction alone.
-    pub gas_used: Option<U256>,
+    #[serde(with = "alloy_serde::u64_hex_opt")]
+    pub gas_used: Option<u64>,
     /// The price paid post-execution by the transaction (i.e. base fee + priority fee). Both
     /// fields in 1559-style transactions are maximums (max fee + max priority fee), the amount
     /// that's actually paid by users can only be determined post-execution
-    pub effective_gas_price: U128,
+    #[serde(with = "alloy_serde::u64_hex_or_decimal")]
+    pub effective_gas_price: u64,
     /// Blob gas used by the eip-4844 transaction
     ///
     /// This is None for non eip-4844 transactions
-    #[serde(skip_serializing_if = "Option::is_none")]
-    pub blob_gas_used: Option<U128>,
+    #[serde(skip_serializing_if = "Option::is_none", with = "alloy_serde::u64_hex_opt", default)]
+    pub blob_gas_used: Option<u64>,
     /// The price paid by the eip-4844 transaction per blob gas.
-    #[serde(skip_serializing_if = "Option::is_none")]
-    pub blob_gas_price: Option<U128>,
+    #[serde(skip_serializing_if = "Option::is_none", with = "alloy_serde::u64_hex_opt", default)]
+    pub blob_gas_price: Option<u64>,
     /// Address of the sender
     pub from: Address,
     /// Address of the receiver. None when its a contract creation transaction.
     pub to: Option<Address>,
     /// Contract address created, or None if not a deployment.
     pub contract_address: Option<Address>,
-    /// Logs emitted by this transaction.
-    pub logs: Vec<Log>,
-    /// Logs bloom
-    pub logs_bloom: Bloom,
     /// The post-transaction stateroot (pre Byzantium)
     ///
     /// EIP98 makes this optional field, if it's missing then skip serializing it
     #[serde(skip_serializing_if = "Option::is_none", rename = "root")]
     pub state_root: Option<B256>,
-    /// Status: either 1 (success) or 0 (failure). Only present after activation of EIP-658
-    #[serde(skip_serializing_if = "Option::is_none", rename = "status")]
-    pub status_code: Option<U64>,
-    /// EIP-2718 Transaction type.
-    ///
-    /// For legacy transactions this returns `0`. For EIP-2718 transactions this returns the type.
-    #[serde(rename = "type")]
-    pub transaction_type: U8,
```

Comment:
is this covered by `ReceiptEnvelope`?

Reply:
yes, it is implicit in the variant. Can add a getter for it if too